pkg/operator/status/status_controller.go

```diff
@@ -89,7 +89,6 @@ func (c StatusSyncer) sync() error {
 			ObjectMeta: metav1.ObjectMeta{Name: c.clusterOperatorName},
 		}
 	}
-	clusterOperatorObj.Status.Conditions = nil
```

Comment:
I agree that clearing always was bad, but shouldn't we ensure that we clean the conditions eventually for operator? We clean properly available and progressing conditions, but I don't think we are clearing failing, shouldn't we?